src/app/timetable/page.tsx

```diff
@@ -0,0 +1,90 @@
+'use client';
+import styles from './styles.module.scss';
+import { useEffect, useState } from 'react';
+import { useAPI } from '@/api/api';
+import { DAY_LENGTH, roundToStartOfDay } from '@/utils/utils';
+import { TimetableDay } from '@/components/timetable/TimetableDay';
+import { format } from 'date-fns';
+import * as locale from 'date-fns/locale';
+import Button from '@/components/UI/Button';
+import { useTimetableEvents } from '@/api/timetable/getTimetableEvents';
+import { createTimetableEvent } from '@/api/timetable/createTimetableEvent';
+
+export default function TimetablePage() {
+  const [firstDay, setFirstDay] = useState(roundToStartOfDay(new Date()));
+  const [numberOfDays, setNumberOfDays] = useState(7);
+  const events = useTimetableEvents(firstDay, numberOfDays);
+  const [eventIndicesPerDay, setEventIndicesPerDay] = useState([] as number[][]);
+  const api = useAPI();
+  useEffect(() => {
+    const offset = numberOfDays === 7 ? -(firstDay.getDay() === 0 ? 6 : firstDay.getDay() - 1) : 0;
```

Comment:
Peut être plus simple à écrire comme ça ?
```suggestion
    const offset = numberOfDays === 7 ? -((firstDay.getDay() + 5) % 6) : 0;
```
Ps : peut être commenter et préciser que le hook sert à se recaler sur le début d'une semaine ?

Reply:
Je sais pas si c'est plus lisible tho